compiler/codegen/cg-expr.cpp

```diff
@@ -1547,15 +1587,9 @@ GenRet createTempVar(Type* t)
     // (to do with references and references pointers)
     // It's not a problem for C since the type will
     // be added to the header before the C compiler runs.
-    GenRet tmp = t;
-    llvm::Type* llTy = tmp.type;
-    INT_ASSERT(llTy);
-
-    llvm::AllocaInst* alloca = createVarLLVM(llTy);
-    llvm::MaybeAlign alignment = getAlignment(t);
-    if (alignment) {
-      alloca->setAlignment(*alignment);
-    }
+    // getLLVMType() code-generates the type, if needed.
+    llvm::Type* llTy = t->symbol->getLLVMType();
+    llvm::AllocaInst* alloca = createVarLLVM(llTy, t, (Symbol*)nullptr);
```

Comment:
Same here, why is the cast needed?

Reply:
In the two cases above, I added the casts to nudge the reader towards understanding that the null goes to the "symbol" vs. the "type" formal argument. If this is problematic, I can remove the casts, as they are not needed for correct operation of the code.